sklearn/metrics/_pairwise_distances_reduction/__init__.py

```diff
@@ -46,56 +46,56 @@
 #
 #
 #                               (base dispatcher)
-#                           PairwiseDistancesReduction
+#                                   Dispatcher
```

Comment:
I find the `Dispatcher` name too generic.

I would rather name this `DistanceReductionDispatcher`.

If want to be explicit that this just a base class that is not meant to be used directly, then `BaseDistanceReductionDispatcher`. It's a bit long but it's not a problem because no other code should not call this class directly.